src/components/navigation-header/index.tsx

```diff
@@ -38,32 +49,58 @@ const MobileMenuButton = () => {
 /**
  * Handles rendering the Sign In and Sign Up UI elements. Automatically hides
  * the elements with CSS on tablet and smaller viewports.
- *
- * @TODO Render user dropdown disclosure for authenticated users
- * ref: https://app.asana.com/0/1202097197789424/1202665629707458/f
  */
 const AuthenticationControls = () => {
-	const { isAuthEnabled, isAuthenticated, isLoading, signIn } =
+	const { showAuthenticatedUI, showUnauthenticatedUI, signIn, signOut, user } =
 		useAuthentication()
-	const shouldShowAuthButtons = isAuthEnabled && !isLoading && !isAuthenticated
 
-	if (!shouldShowAuthButtons) {
+	if (!showAuthenticatedUI && !showUnauthenticatedUI) {
 		return null
 	}
 
-	return (
-		<div className={s.authenticationControls}>
-			<Button onClick={() => signIn()} text="Sign In" />
-			<StandaloneLink
-				className={s.signUpLink}
-				textClassName={s.signUpLinkText}
-				href="/sign-up"
-				icon={<IconUserPlus16 />}
-				iconPosition="trailing"
-				text="Sign Up"
+	let content
+	if (showUnauthenticatedUI) {
+		content = (
+			<>
+				<Button onClick={() => signIn()} text="Sign In" />
+				<StandaloneLink
+					className={s.signUpLink}
+					textClassName={s.signUpLinkText}
+					href="/sign-up"
+					icon={<IconUserPlus16 />}
+					iconPosition="trailing"
+					text="Sign Up"
+				/>
+			</>
+		)
+	} else if (showAuthenticatedUI) {
+		content = (
+			<UserDropdownDisclosure
+				className={s.userDropdownDisclosure}
+				listPosition="right"
+				items={[
+					{
+						href: '/bookmarks',
+						icon: <IconBookmark16 />,
+						label: 'Bookmarks',
+					},
+					{
+						href: 'https://portal.cloud.hashicorp.com/account-settings',
+						icon: <IconExternalLink16 />,
+						label: 'Account Settings',
```

Comment:
💭 We probably want this link to open a new tab? If it's going to the external HCP settings page.

Reply:
I was wondering about this too. It'll require some refactoring so I'll do it outside of this PR, but I've taken a note to ask design folks about this! 😊